cnd/src/comit_api.rs

```diff
@@ -14,26 +14,26 @@ use std::str::FromStr;
 #[derive(Clone, Copy, derivative::Derivative, PartialEq)]
 #[derivative(Debug = "transparent")]
 pub enum LedgerKind {
-    BitcoinMainnet,
-    BitcoinTestnet,
-    BitcoinRegtest,
-    Ethereum(Ethereum),
+    Bitcoin(ledger::Bitcoin),
+    Ethereum(ledger::Ethereum),
 }
 
 impl FromHeader for LedgerKind {
     fn from_header(mut header: Header) -> Result<Self, serde_json::Error> {
         Ok(match header.value::<String>()?.as_str() {
             "bitcoin" => match header.take_parameter::<String>("network")?.as_ref() {
-                "mainnet" => LedgerKind::BitcoinMainnet,
-                "testnet" => LedgerKind::BitcoinTestnet,
-                "regtest" => LedgerKind::BitcoinRegtest,
+                "mainnet" => LedgerKind::Bitcoin(ledger::Bitcoin::Mainnet),
```

Comment:
Then these would become `network::Bitcoin::Mainnet` - and walah now I remember why this doesn't work, we cannot have another 'network' module in the root of cnd. Perhaps we need to submodule all the ledger stuff (e.g., identity, transaction, htlc_location, and now - network)?

Reply:
I would see it as ledger specific: `bitcoin::Network::Mainnet`.
For Ethereum, the term is chain, so it would be: `ethereum::Chain::Ropsten`.